src/v/serde/test/serde_test.cc

```diff
@@ -1077,3 +1079,62 @@ SEASTAR_THREAD_TEST_CASE(collections_interop) {
     BOOST_REQUIRE(serialized_vector == serialized_fifo);
     BOOST_REQUIRE(serialized_vector == serialized_f_vector);
 }
+
+template<typename T>
+concept IsSerdeReadable = requires(
+  iobuf_parser& in, T& t, const std::size_t bytes_left_limit) {
+    { serde::tag_invoke(serde::read_tag, in, t, bytes_left_limit) };
+};
+
+template<typename T>
+concept IsSerdeWritable = requires(T&& t, iobuf b) {
+    { serde::tag_invoke(serde::write_tag, b, std::forward<T>(t)) };
+};
+
+SEASTAR_THREAD_TEST_CASE(variant) {
+    using my_variant = serde::variant<ss::sstring, int, bool, my_enum, int>;
```

Comment:
how about a test case that uses `std::monostate`?

Reply:
std::monostate is not supported by serde, should I add support for it?